CCFatigue_modules/2_S-NCurves/sn_curve_linlog.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python
+""" CCFatigue - Module 2 - sn_curve_linlog.py
+This code takes in AGG data and output SNC data
+"""
+
+import os
+import math
+import numpy as np
+import pandas as pd
+from scipy import stats
+from itertools import chain
+
+from astm import Astm
+
+
+CONFIDENCE = 95
+RELIABILITY_LEVEL = 50
+
+SRC_DIR = os.path.dirname(os.path.realpath(__file__))
+DATA_DIR = os.path.join(SRC_DIR, "..", "..", "Data")
+
+
+INPUT_FILENAME = "AGG_input.csv"
+INPUT_FILE = os.path.join(DATA_DIR, INPUT_FILENAME)
+OUTPUT_JSON_FILENAME = "SNC_LinLog.json"
+OUTPUT_JSON_FILE = os.path.join(DATA_DIR, OUTPUT_JSON_FILENAME)
+OUTPUT_CSV_FILENAME = "SNC_LinLog.csv"
+OUTPUT_CSV_FILE = os.path.join(DATA_DIR, OUTPUT_CSV_FILENAME)
+
+
+def equation_ycar(slope: float, intercept: float, stress_parameter: float) -> float:
+    """
+    https://github.com/EPFL-ENAC/CCFatiguePlatform/blob/develop/CCFatigue_modules/2_S-NCurves/S-N-Curve-LinLog.for#L314
+    Inputs:
+    - slope
+    - intercept
+    - stress_parameter
+    Outputs:
+    - ycar
+    """
+    ycar = slope + intercept * stress_parameter
+    return ycar
+
+
+def stress_at_failure(aa: float, bb: float, nn: float) -> float:
+    """
+    Calculate stress at failure
+    https://github.com/EPFL-ENAC/CCFatiguePlatform/blob/develop/CCFatigue_modules/2_S-NCurves/S-N-Curve-LinLog.for#L397
+    Inputs:
+    - aa
+    - bb
+    - nn
+    Outputs:
+    - spn: stress at failure
+    """
+    spn = aa + math.log10(nn) * bb
+    return spn
+
+
+def stress_at_failure_bounds(
+    sample_count: int, q: float, a: float, b: float, nn: float, pp: float, xb: float
+) -> tuple[float, float]:
+    """Get Stress at failure (sigma_max)
+    returns (lower bound, upper bound)"""
+
+    term1 = (
+        (sample_count * q * a * b)
+        - (sample_count * q * math.log10(nn) * b)
+        + (pp * sample_count * xb)
+    )
+    term2 = math.sqrt(
+        sample_count
+        * pp
+        * q
+        * (
+            2 * sample_count * a * b * xb
+            - 2 * sample_count * math.log10(nn) * b * xb
+            + q * b**2
+            + sample_count * b**2 * xb**2
+            + sample_count * math.log10(nn) ** 2
+            - 2 * sample_count * math.log10(nn) * a
+            - pp
+            + sample_count * a**2
+        )
+    )
+    term3 = sample_count * (b**2 * q - pp)
+
+    stress_upperbound = -(term1 + term2) / term3
+    stress_lowerbound = -(term1 - term2) / term3
+
+    return (stress_lowerbound, stress_upperbound)
+
+
+if __name__ == "__main__":
+
+    astm = Astm(CONFIDENCE)
+
+    # Import input file (AGG format)
+    agg_df = pd.read_csv(INPUT_FILE)
+
+    # Add columns with log10(stress_parameter) and log10(cycles_to_failure)
+    # agg_df["log10_stress_parameter"] = np.log10(agg_df.stress_parameter)

```

Comment:
Is this line needed or not ? If yes, should not be commented out, if not, description line 101 should be adjusted